<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-07">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="beta-diversity-metrics.html">
<link rel="next" href="inference-of-microbial-ecological-networks.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="open-reproducible-microbiome-data-analysis-spring-school-2018.html"><a href="open-reproducible-microbiome-data-analysis-spring-school-2018.html"><i class="fa fa-check"></i><b>1</b> OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>2.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#useful-functions"><i class="fa fa-check"></i><b>2.2</b> Useful functions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>3</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="3.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>3.1</b> Structure</a></li>
<li class="chapter" data-level="3.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>3.2</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="3.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>3.3</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="3.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>3.4</b> Read the tree file.</a></li>
<li class="chapter" data-level="3.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>3.5</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="3.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-test-data"><i class="fa fa-check"></i><b>3.6</b> Read test data</a></li>
<li class="chapter" data-level="3.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#pre-processing-data-check"><i class="fa fa-check"></i><b>3.7</b> Pre-processing data check</a><ul>
<li class="chapter" data-level="3.7.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#sequencing-depth"><i class="fa fa-check"></i><b>3.7.1</b> Sequencing depth</a></li>
<li class="chapter" data-level="3.7.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#distribution-of-otus"><i class="fa fa-check"></i><b>3.7.2</b> Distribution of OTUs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>4</b> Alpha diversities</a><ul>
<li class="chapter" data-level="4.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>4.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="4.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>4.2</b> Diversities</a><ul>
<li class="chapter" data-level="4.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>4.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="4.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>4.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>5</b> Composition plots</a><ul>
<li class="chapter" data-level="5.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>5.1</b> Barplot counts</a></li>
<li class="chapter" data-level="5.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>5.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="5.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>5.2.1</b> Barplot customize</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="composition-plots.html"><a href="composition-plots.html#heatmaps"><i class="fa fa-check"></i><b>5.3</b> Heatmaps</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>6</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="6.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>6.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="6.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>6.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="6.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>6.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>6.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="6.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>6.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="6.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>6.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>7</b> Core microbiota</a><ul>
<li class="chapter" data-level="7.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>7.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="7.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>7.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="7.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>7.3</b> Core visualization</a><ul>
<li class="chapter" data-level="7.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>7.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>8</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="8.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>8.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="8.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>8.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="8.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>8.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="8.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>8.2.2</b> igraph network</a></li>
<li class="chapter" data-level="8.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>8.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>8.3</b> Network properties</a><ul>
<li class="chapter" data-level="8.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>8.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="8.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>8.3.2</b> Network plot</a></li>
<li class="chapter" data-level="8.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>8.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="core-microbiota" class="section level1">
<h1><span class="header-section-number">7</span> Core microbiota</h1>
<p>For more information:</p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S1198743X14609629?via%3Dihub">The adult intestinal core microbiota is determined by analysis depth and health status</a>.</p>
<p><a href="https://academic.oup.com/femsre/article/41/2/182/2979411">Intestinal microbiome landscaping: insight in community assemblage and implications for microbial modulation strategies</a>.</p>
<p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0023035">Intestinal Microbiota in Healthy Adults: Temporal Analysis Reveals Individual and Common Core and Relation to Intestinal Symptoms</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(microbiome) <span class="co"># data analysis and visualisation</span>
<span class="kw">library</span>(phyloseq) <span class="co"># also the basis of data object. Data analysis and visualisation</span>
<span class="kw">library</span>(RColorBrewer) <span class="co"># nice color options</span>
<span class="kw">library</span>(ggpubr) <span class="co"># publication quality figures, based on ggplot2</span>
<span class="kw">library</span>(dplyr) <span class="co"># data handling  </span></code></pre></div>
<div id="core-microbiota-anlaysis" class="section level2">
<h2><span class="header-section-number">7.1</span> Core microbiota anlaysis</h2>
<p>We will use the filtered phyloseq object from previous tutorial.We will use the filtered phyloseq object from the Microbiome Analytics Workshop 2018 tutorial I.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># read non rarefied data</span>
ps1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;./phyobjects/ps1.rds&quot;</span>)

<span class="co"># use print option to see the data saved as phyloseq object.</span></code></pre></div>
<p>Subset the data to keep only stool samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.stool &lt;-<span class="st"> </span><span class="kw">subset_samples</span>(ps1, bodysite ==<span class="st"> &quot;Stool&quot;</span>)

<span class="co"># convert to relative abundance  </span>
ps1.stool.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(ps1.stool, <span class="st">&quot;compositional&quot;</span>)
<span class="kw">print</span>(ps1.stool.rel)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 3690 taxa and 169 samples ]
## sample_data() Sample Data:       [ 169 samples by 31 sample variables ]
## tax_table()   Taxonomy Table:    [ 3690 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 3690 tips and 3689 internal nodes ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ps1.stool.rel2 &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(ps1.stool.rel) &gt;<span class="st"> </span><span class="dv">0</span>, ps1.stool.rel)

<span class="kw">print</span>(ps1.stool.rel2)</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 2717 taxa and 169 samples ]
## sample_data() Sample Data:       [ 169 samples by 31 sample variables ]
## tax_table()   Taxonomy Table:    [ 2717 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 2717 tips and 2716 internal nodes ]</code></pre>
<p>Check of the core OTUs</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">core.taxa.standard &lt;-<span class="st"> </span><span class="kw">core_members</span>(ps1.stool.rel2, <span class="dt">detection =</span> <span class="fl">0.001</span>, <span class="dt">prevalence =</span> <span class="dv">50</span>/<span class="dv">100</span>)

<span class="kw">print</span>(core.taxa.standard)</code></pre></div>
<pre><code>##  [1] &quot;191483&quot;  &quot;194909&quot;  &quot;197214&quot;  &quot;177343&quot;  &quot;184753&quot;  &quot;560336&quot;  &quot;192684&quot; 
##  [8] &quot;535375&quot;  &quot;193233&quot;  &quot;772282&quot;  &quot;4325533&quot; &quot;850586&quot;  &quot;196219&quot;  &quot;180082&quot; 
## [15] &quot;330458&quot;  &quot;173996&quot;</code></pre>
<p>There are 16 OTUs that are core based on the cut-offs for prevalence and detection we choose. However, we only see IDs, not very informative. We can get the classification of these as below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Extract the taxonomy table</span>

taxonomy &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">tax_table</span>(ps1.stool.rel2))

<span class="co"># Subset this taxonomy table to include only core OTUs  </span>
core_taxa_id &lt;-<span class="st"> </span><span class="kw">subset</span>(taxonomy, <span class="kw">rownames</span>(taxonomy) %in%<span class="st"> </span>core.taxa.standard)

DT::<span class="kw">datatable</span>(core_taxa_id)</code></pre></div>
<div id="htmlwidget-7cef22cf2f4be7d85fc4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7cef22cf2f4be7d85fc4">{"x":{"filter":"none","data":[["191483","194909","197214","177343","184753","560336","192684","535375","193233","772282","4325533","850586","196219","180082","330458","173996"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Bacteroidetes","Firmicutes","Firmicutes"],["Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Clostridia","Clostridia"],["Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Bacteroidales","Clostridiales","Clostridiales"],["Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Bacteroidaceae","Rikenellaceae","Rikenellaceae","[Odoribacteraceae]","Porphyromonadaceae","Porphyromonadaceae","Lachnospiraceae","Lachnospiraceae"],["Bacteroides","Bacteroides","Bacteroides","Bacteroides","Bacteroides","Bacteroides","Bacteroides","Bacteroides","Bacteroides",null,null,"Odoribacter","Parabacteroides","Parabacteroides",null,null],[null,null,null,null,null,null,null,"ovatus",null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="core-abundance-and-diversity" class="section level2">
<h2><span class="header-section-number">7.2</span> Core abundance and diversity</h2>
<p>Total core abundance in each sample (sum of abundances of the core members):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">core.abundance &lt;-<span class="st"> </span><span class="kw">sample_sums</span>(<span class="kw">core</span>(ps1.stool.rel2, <span class="dt">detection =</span> <span class="fl">0.001</span>, <span class="dt">prevalence =</span> <span class="dv">50</span>/<span class="dv">100</span>))

DT::<span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(core.abundance))</code></pre></div>
<div id="htmlwidget-ab39ac141f1094a475b3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ab39ac141f1094a475b3">{"x":{"filter":"none","data":[["1927.SRS014659.SRX020511.SRR045649","1927.SRS015854.SRX020554.SRR043727","1927.SRS015599.SRX020539.SRR044148","1927.SRS015452.SRX020545.SRR044740","1927.SRS014442.SRX020511.SRR045656","1927.SRS014287.SRX020511.SRR045664","1927.SRS049428.SRX020670.SRR047663","1927.SRS019910.SRX020516.SRR044347","1927.SRS050422.SRX020511.SRR045628","1927.SRS064276.SRX020670.SRR046347","1927.SRS052697.SRX020550.SRR046441","1927.SRS063827.SRX020532.SRR046984","1927.SRS062847.SRX020532.SRR047036","1927.SRS014659.SRX020543.SRR045556","1927.SRS055482.SRX020550.SRR047716","1927.SRS021910.SRX020675.SRR047783","1927.SRS011271.SRX020661.SRR044827","1927.SRS011529.SRX020659.SRR044994","1927.SRS011271.SRX020659.SRR045051","1927.SRS018733.SRX020516.SRR047425","1927.SRS015281.SRX020554.SRR043805","1927.SRS011415.SRX020661.SRR044786","1927.SRS014287.SRX020543.SRR045558","1927.SRS046382.SRX020540.SRR047262","1927.SRS064321.SRX020519.SRR046501","1927.SRS056259.SRX020578.SRR047235","1927.SRS020470.SRX022097.SRR057663","1927.SRS015247.SRX020577.SRR044079","1927.SRS015724.SRX020539.SRR044132","1927.SRS015782.SRX020539.SRR044139","1927.SRS018872.SRX020572.SRR051576","1927.SRS016203.SRX020516.SRR044343","1927.SRS019089.SRX020564.SRR044731","1927.SRS011621.SRX020661.SRR044776","1927.SRS011413.SRX020661.SRR044792","1927.SRS011653.SRX020659.SRR044997","1927.SRS011413.SRX020659.SRR045020","1927.SRS019582.SRX020543.SRR045562","1927.SRS050422.SRX020543.SRR045575","1927.SRS048819.SRX020511.SRR045612","1927.SRS054461.SRX020550.SRR046442","1927.SRS042703.SRX020540.SRR047267","1927.SRS046639.SRX020670.SRR047659","1927.SRS023047.SRX020675.SRR047826","1927.SRS016056.SRX020536.SRR049357","1927.SRS065725.SRX020519.SRR049530","1927.SRS011410.SRX020661.SRR044800","1927.SRS014442.SRX020543.SRR045553","1927.SRS048819.SRX020543.SRR045568","1927.SRS045877.SRX020523.SRR046406","1927.SRS057447.SRX020523.SRR046411","1927.SRS046313.SRX020550.SRR046437","1927.SRS014613.SRX020546.SRR047386","1927.SRS021601.SRX020675.SRR047748","1927.SRS022987.SRX020675.SRR047766","1927.SRS022924.SRX020675.SRR047811","1927.SRS020176.SRX020676.SRR047859","1927.SRS024511.SRX020676.SRR047887","1927.SRS063807.SRX022232.SRR058094","1927.SRS063968.SRX022156.SRR058115","1927.SRS014823.SRX020546.SRR043699","1927.SRS018712.SRX020572.SRR046003","1927.SRS014885.SRX020554.SRR043822","1927.SRS014369.SRX020554.SRR043823","1927.SRS065466.SRX020532.SRR047056","1927.SRS050733.SRX020550.SRR047710","1927.SRS048870.SRX020577.SRR044064","1927.SRS015578.SRX020577.SRR044021","1927.SRS016018.SRX020539.SRR044156","1927.SRS015911.SRX020520.SRR044615","1927.SRS019089.SRX020545.SRR044757","1927.SRS011452.SRX020661.SRR044798","1927.SRS011586.SRX020661.SRR044826","1927.SRS053335.SRX020531.SRR046056","1927.SRS043411.SRX020670.SRR046345","1927.SRS045526.SRX020670.SRR046355","1927.SRS043299.SRX020523.SRR046410","1927.SRS049823.SRX020523.SRR046414","1927.SRS045607.SRX020523.SRR046421","1927.SRS063275.SRX020532.SRR046899","1927.SRS049164.SRX020540.SRR047273","1927.SRS042628.SRX020578.SRR047303","1927.SRS045414.SRX020563.SRR047697","1927.SRS015389.SRX020546.SRR043661","1927.SRS015190.SRX020554.SRR043731","1927.SRS019787.SRX020577.SRR043974","1927.SRS015332.SRX020577.SRR044071","1927.SRS015911.SRX020510.SRR044585","1927.SRS019013.SRX020510.SRR044597","1927.SRS019013.SRX020520.SRR044627","1927.SRS011410.SRX020659.SRR045027","1927.SRS048083.SRX020677.SRR047956","1927.SRS023488.SRX020678.SRR048028","1927.SRS050599.SRX020678.SRR048036","1927.SRS065665.SRX020519.SRR049404","1927.SRS016267.SRX020577.SRR044054","1927.SRS011452.SRX020659.SRR047485","1927.SRS012191.SRX020675.SRR047747","1927.SRS016095.SRX020516.SRR044317","1927.SRS014999.SRX020577.SRR044050","1927.SRS011653.SRX020661.SRR044770","1927.SRS019582.SRX020511.SRR045623","1927.SRS018559.SRX020516.SRR044352","1927.SRS019267.SRX020516.SRR044360","1927.SRS015960.SRX020577.SRR044087","1927.SRS063324.SRX020519.SRR046503","1927.SRS054590.SRX020550.SRR046427","1927.SRS011157.SRX020659.SRR045036","1927.SRS015663.SRX020539.SRR044161","1927.SRS020119.SRX020676.SRR047844","1927.SRS015815.SRX020562.SRR044705","1927.SRS046349.SRX020531.SRR046051","1927.SRS015452.SRX020564.SRR047456","1927.SRS011529.SRX020661.SRR044768","1927.SRS011157.SRX020661.SRR044808","1927.SRS011415.SRX020659.SRR045013","1927.SRS011586.SRX020659.SRR045050","1927.SRS018920.SRX020572.SRR045810","1927.SRS043804.SRX020531.SRR046050","1927.SRS052326.SRX020670.SRR046232","1927.SRS054928.SRX020670.SRR046262","1927.SRS065263.SRX020670.SRR046351","1927.SRS055563.SRX020523.SRR046412","1927.SRS042387.SRX020523.SRR046416","1927.SRS013543.SRX020677.SRR047980","1927.SRS064557.SRX020569.SRR046542","1927.SRS063961.SRX020532.SRR046907","1927.SRS062464.SRX020532.SRR046982","1927.SRS063214.SRX020532.SRR046986","1927.SRS051031.SRX020578.SRR047244","1927.SRS042669.SRX020540.SRR047279","1927.SRS058416.SRX020523.SRR047701","1927.SRS055665.SRX020523.SRR047702","1927.SRS049748.SRX020523.SRR047703","1927.SRS021541.SRX020675.SRR047797","1927.SRS013762.SRX020677.SRR048007","1927.SRS019381.SRX020572.SRR051585","1927.SRS021109.SRX022097.SRR057663","1927.SRS020584.SRX022097.SRR057663","1927.SRS020641.SRX022097.SRR057663","1927.SRS042803.SRX022156.SRR058115","1927.SRS063918.SRX022232.SRR058094","1927.SRS064647.SRX022232.SRR058094","1927.SRS049766.SRX022241.SRR058097","1927.SRS056832.SRX022241.SRR058097","1927.SRS063608.SRX022156.SRR058115","1927.SRS011621.SRX020659.SRR045004","1927.SRS011405.SRX020659.SRR046237","1927.SRS015518.SRX020539.SRR044146","1927.SRS011405.SRX020661.SRR044803","1927.SRS018968.SRX020572.SRR045817","1927.SRS063921.SRX020569.SRR046605","1927.SRS023422.SRX020683.SRR048341","1927.SRS019534.SRX020572.SRR051591","1927.SRS021853.SRX022232.SRR058094","1927.SRS065421.SRX022232.SRR058094","1927.SRS063307.SRX022232.SRR058094","1927.SRS042572.SRX022241.SRR058097","1927.SRS049949.SRX022241.SRR058097","1927.SRS057676.SRX022156.SRR058115","1927.SRS014923.SRX020546.SRR043675","1927.SRS051892.SRX022156.SRR058115","1927.SRS048838.SRX020523.SRR046418","1927.SRS057122.SRX020677.SRR047969","1927.SRS014345.SRX020546.SRR047368","1927.SRS023851.SRX020676.SRR047925","1927.SRS046341.SRX020554.SRR043828","1927.SRS023422.SRX020676.SRR047874","1927.SRS020811.SRX022097.SRR057663"],[0.0465431540894713,0.401378326996198,0.282029950083195,0.392269280562234,0.567991933160472,0.4,0.40018176310209,0.249308118081181,0.464148033924441,0.36194221234399,0.260434569113675,0.233664772727273,0.421884317233154,0.0711857203569911,0.172183047920564,0.255957634598411,0.146615430091635,0.102931870913309,0.135868120180803,0.00842696629213483,0.24501708428246,0.20120572720422,0.3818634313268,0.473793512658228,0.411666357049972,0.361031518624642,0.367285499247366,0.234413407821229,0.456430666108085,0.278208278208278,0.280252764612954,0.525851938895417,0.341131534972547,0.38109756097561,0.323096129837703,0.0810419681620839,0.316417212347989,0.230706742485784,0.429861529199278,0.469484629294756,0.218311213694941,0.354477180820335,0.403169014084507,0.224445812807882,0.405931242916509,0.600860178782257,0.273193234238852,0.517786069651741,0.39090523937297,0.362703467799009,0.315261789825473,0.714163090128755,0.323939881910896,0.247454751131222,0.389038031319911,0.367110887603721,0.265678449258837,0.70744935396395,0.491539970573811,0.22855773838045,0.477044694435999,0.487407407407407,0.323519682434668,0.210704862419902,0.403456048084147,0.11132712575463,0.47650827549386,0.52166776099803,0.374000761324705,0.219600725952813,0.326815642458101,0.187238723872387,0.441889763779528,0.445383912248629,0.588142178730753,0.415516932449382,0.42611060743427,0.28087044534413,0.576525821596244,0.157524322086525,0.234841035725991,0.26274328081557,0.158908822806586,0.388976377952756,0.671440606571188,0.296296296296296,0.302752293577982,0.225370418507928,0.258557660352277,0.252818585582508,0.270007124198528,0.270133936336754,0.396285979572888,0.527178404186943,0.0671641791044776,0.139395338102931,0.217722534081796,0.157066189624329,0.179702048417132,0.187580437580438,0.0859349855165755,0.24083269671505,0.403130613292276,0.799706529713866,0.332867458653622,0.421744767549546,0.00679953106682298,0.158000574547544,0.151909150757077,0.374622356495468,0.182673942701228,0.173104850421144,0.381830297645491,0.102909713307659,0.166075650118203,0.184064237183447,0.446985446985447,0.118605315804516,0.45103598691385,0.787892113749634,0.460692951015532,0.361114745518775,0.463103122043519,0.104176668266923,0.383466304532337,0.584875040571243,0.337055606198724,0.190838741887169,0.113517745302714,0.285714285714286,0.13189750039302,0.458975940777298,0.304897314375987,0.153448275862069,0.252950643776824,0.779561539789401,0.625614339404452,0.54324942791762,0.376998050682261,0.633716475095785,0.281361097003555,0.511503602138043,0.216030534351145,0.608873505349276,0.233232366940718,0.159851301115242,0.386591962905719,0.466666666666667,0.606121537086685,0.445582281495594,0.182204590535562,0.94693787309308,0.661422708618331,0.632988435788192,0.432082794307891,0.432393693263258,0.0233545647558386,0.24816280384398,0.245683930942895,0.439284485724114,0.0395156150414277,0.512964563526361,0.429738101943115,0.109796784956021,0.536155666579017,0.37303970637304,0.173579801623084,0.759483770043019,0.120355411954766]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>core.abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="core-visualization" class="section level2">
<h2><span class="header-section-number">7.3</span> Core visualization</h2>
<div id="core-heatmaps" class="section level3">
<h3><span class="header-section-number">7.3.1</span> Core heatmaps</h3>
<p>This visualization method has been used for instance in <a href="https://academic.oup.com/femsre/article/41/2/182/2979411">Intestinal microbiome landscaping: insight in community assemblage and implications for microbial modulation strategies</a>.</p>
<p>Note that you can order the taxa on the heatmap with the order.taxa argument.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Core with compositionals:</span>
prevalences &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">05</span>, <span class="dv">1</span>, .<span class="dv">05</span>)
detections &lt;-<span class="st"> </span><span class="dv">10</span>^<span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-3</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)

<span class="co"># Also define gray color palette</span>
gray &lt;-<span class="st"> </span><span class="kw">gray</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="dv">5</span>))
p.core &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.stool.rel2, <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, <span class="dt">colours =</span> gray,
    <span class="dt">prevalences =</span> prevalences, <span class="dt">detections =</span> detections, <span class="dt">min.prevalence =</span> .<span class="dv">5</span>) +
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)
<span class="kw">print</span>(p.core)    </code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Same with the viridis color palette</span>
<span class="co"># color-blind friendly and uniform</span>
<span class="co"># options: viridis, magma, plasma, inferno</span>
<span class="co"># https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html</span>
<span class="co"># Also discrete=TRUE versions available</span>
<span class="kw">library</span>(viridis)</code></pre></div>
<pre><code>## Loading required package: viridisLite</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(p.core +<span class="st"> </span><span class="kw">scale_fill_viridis</span>())</code></pre></div>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-7-2.png" width="672" /></p>
<p>Color change</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Core with compositionals:</span>
prevalences &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">05</span>, <span class="dv">1</span>, .<span class="dv">05</span>)
detections &lt;-<span class="st"> </span><span class="dv">10</span>^<span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-3</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)

<span class="co"># Also define gray color palette</span>
gray &lt;-<span class="st"> </span><span class="kw">gray</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="dv">5</span>))
p.core &lt;-<span class="st"> </span><span class="kw">plot_core</span>(ps1.stool.rel2, 
                    <span class="dt">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, 
                    <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),
                    <span class="dt">prevalences =</span> prevalences, 
                    <span class="dt">detections =</span> detections, 
                    <span class="dt">min.prevalence =</span> .<span class="dv">5</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)

<span class="kw">print</span>(p.core) </code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>We have a custom script to format this figure which can be found in <code>scripts</code> folder in the <code>RProject</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;./scripts/plot_core_assist.R&quot;</span>)
plot_data &lt;-<span class="st"> </span>p.core$data

df_plot &lt;-<span class="st"> </span><span class="kw">plot_core_assist</span>(plot_data, <span class="kw">tax_table</span>(ps1.stool.rel2), <span class="dt">levels =</span> <span class="st">&quot;Family&quot;</span>)

p.core$data &lt;-<span class="st"> </span>df_plot

<span class="kw">plot</span>(p.core +<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&quot;italic&quot;</span>)))</code></pre></div>
<p><img src="06-MAW-PV_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.4 (2018-03-15)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 7 x64 (build 7601) Service Pack 1
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=Dutch_Netherlands.1252  LC_CTYPE=Dutch_Netherlands.1252   
## [3] LC_MONETARY=Dutch_Netherlands.1252 LC_NUMERIC=C                      
## [5] LC_TIME=Dutch_Netherlands.1252    
## 
## attached base packages:
## [1] methods   stats     graphics  grDevices utils     datasets  base     
## 
## other attached packages:
##  [1] bindrcpp_0.2.2     viridis_0.5.1      viridisLite_0.3.0 
##  [4] dplyr_0.7.4        ggpubr_0.1.6       magrittr_1.5      
##  [7] RColorBrewer_1.1-2 microbiome_1.0.2   ggplot2_2.2.1     
## [10] phyloseq_1.23.1   
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.38.0      tidyr_0.8.0         jsonlite_1.5       
##  [4] splines_3.4.4       foreach_1.4.4       shiny_1.0.5        
##  [7] assertthat_0.2.0    stats4_3.4.4        yaml_2.1.18        
## [10] pillar_1.2.2        backports_1.1.2     lattice_0.20-35    
## [13] glue_1.2.0          digest_0.6.15       promises_1.0.1     
## [16] XVector_0.18.0      colorspace_1.3-2    htmltools_0.3.6    
## [19] httpuv_1.4.1        Matrix_1.2-14       plyr_1.8.4         
## [22] pkgconfig_2.0.1     bookdown_0.7        zlibbioc_1.24.0    
## [25] purrr_0.2.4         xtable_1.8-2        scales_0.5.0       
## [28] later_0.7.1         tibble_1.4.2        mgcv_1.8-23        
## [31] IRanges_2.12.0      DT_0.4              BiocGenerics_0.24.0
## [34] lazyeval_0.2.1      survival_2.42-3     mime_0.5           
## [37] evaluate_0.10.1     nlme_3.1-137        MASS_7.3-49        
## [40] vegan_2.5-1         tools_3.4.4         data.table_1.10.4-3
## [43] stringr_1.3.0       S4Vectors_0.16.0    munsell_0.4.3      
## [46] cluster_2.0.7-1     Biostrings_2.46.0   ade4_1.7-11        
## [49] compiler_3.4.4      rlang_0.2.0         rhdf5_2.22.0       
## [52] grid_3.4.4          iterators_1.0.9     biomformat_1.7.0   
## [55] htmlwidgets_1.2     crosstalk_1.0.0     igraph_1.2.1       
## [58] labeling_0.3        rmarkdown_1.9       gtable_0.2.0       
## [61] codetools_0.2-15    multtest_2.34.0     reshape2_1.4.3     
## [64] R6_2.2.2            gridExtra_2.3       knitr_1.20         
## [67] bindr_0.1.1         rprojroot_1.3-2     permute_0.9-4      
## [70] ape_5.1             stringi_1.1.7       parallel_3.4.4     
## [73] Rcpp_0.12.16        tidyselect_0.2.4    xfun_0.1</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="beta-diversity-metrics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="inference-of-microbial-ecological-networks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
